test/NuGet.Core.Tests/NuGet.Common.Test/SynchronizationTest.cs

```diff
@@ -56,21 +53,19 @@ public async Task ConcurrencyUtilities_WaitAcquiresLock()
         }
 
         [Fact]
-        public async Task ConcurrencyUtilities_ZeroTimeoutStillGetsLock()
+        public async Task ConcurrencyUtilities_CancelledTokeDoesNotGetLock()
```

Comment:
This is a change in behaviour, but the scenario doesn't make sense to me: Depending on getting the lock, even though the cancellation token requests cancellation, because you know you're the only thread trying to get the lock. However, `ConcurrencyUtilities` is used across process boundaries, so even if you know you're the only thread in the process getting the lock, and you want to prevent other threads in the process entering until some work is done, it's impossible to guarantee that you're the only process trying to get the lock. Therefore there is no scenario where it makes sense to depend on getting the lock despite asking for cancellation.

It passed in my previous commits because the test itself had a timing bug.